graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/DmlFetcher.java

```diff
@@ -149,6 +158,33 @@ protected Parameters getDatastoreParameters(DataFetchingEnvironment environment)
     return ImmutableMap.of("value", finalValue, "applied", applied);
   }
 
+  protected boolean containsDirective(OperationDefinition operation, String atomicDirective) {
+    return operation.getDirectives().stream().anyMatch(d -> d.getName().equals(atomicDirective));
+  }
+
+  /**
+   * Executes the query in an async way in a fire and forget fashion. Completes immediately with
+   * applied=true without waiting for the result.
+   */
+  protected CompletableFuture<Map<String, Object>> executeAsyncWithoutResult(
+      DataStore dataStore, BoundQuery query, Object originalValue) {
+    dataStore
+        .execute(query)
+        .whenComplete(
+            (r, throwable) -> {
+              if (throwable != null) {
+                log.warn(
+                    String.format(
+                        "The query %s executed within the %s directive failed.",
+                        query, ASYNC_DIRECTIVE),
+                    throwable);
+              }
+            });
+    // complete immediately with applied=true without waiting for the result
+    return CompletableFuture.completedFuture(
+        toAppliedMutationResultWithOriginalValue(originalValue));
```

Comment:
Well, since `@async` annotates specific mutations, we should be able to use a different return type. Put it another way, making a mutation async is a schema change from the client's perspective, so it should not be a surprise the user that the return value is different. WDYT?

Reply:
As far as I can see, we don't have the knowledge of the semantic of the GraphQL query at the `GraphqlResourceBase` where we decide about the HTTP response codes. Also, all responses all returned with the standard `200` code.
We may be able to refactor the whole `GraphQL` worklow to give the `GraphqlResourceBase`:
```
   .thenApply(result -> asyncResponse.resume(result.toSpecification()))
```
feedback about whether it should return `202` or `200` but I am not sure if this is worth the effort (@olim7t wdyt?)
I think that a good middle ground is returning `accepted` in the GraphQL response body as @dimas-b suggested.

Reply:
Yeah, I think the GraphQL infrastructure takes over HTTP code generation, so it would probably be an overkill for us to try to override the HTTP code in this PR.